apps/foundation-demo/src/app/checkout/components/shipping/shipping-form/shipping-form.component.spec.ts

```diff
@@ -3,14 +3,25 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { ShippingFormComponent } from './shipping-form.component';
 import { FormsModule, ReactiveFormsModule, FormControl, FormGroup, AbstractControl } from '@angular/forms';
 import { By } from '@angular/platform-browser';
-import { Component, Input } from '@angular/core';
-import { DaffodilAddress } from '@daffodil/core';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { DaffodilAddress, ShippingOption } from '@daffodil/core';
 import { ErrorStateMatcher } from '../../../../design/molecules/error-state-matcher/error-state-matcher.component';
 
-@Component({'template': '<shipping-form [shippingInfo]="shippingInfoValue" (updateShippingInfo)="onUpdateShippingInfoFunction($event)"></shipping-form>'})
+@Component({'template': '<shipping-form [shippingInfo]="shippingInfoValue" [selectedShippingOption]="selectedShippingOptionValue" [hideContinueToPayment]="hideContinueToPaymentValue" (updateShippingInfo)="onUpdateShippingInfoFunction($event)" (selectShippingOption)="selectShippingOptionFunction($event)" (continueToPayment)="continueToPaymentFunction()"></shipping-form>'})
```

Comment:
I disagree that this API is absurd. There are pros and cons of having the extra inputs and outputs, and there are pros and cons of not having them. Here are my thoughts:

`hidden` - not part of the API, so I'm ignoring it.

`shippingInfo` - equivalent of "data" in your example.

`hideContinueToPayment` - this could be removed, but the form would need to subscribe directly to `showPaymentView`. The form needs to know when it has been submitted for the first time, because of the way the payment form is hidden until the shipping form is submitted. The only alternative I can think of is to change the shipping form submit button to something more generic than "Continue to Payment", like "Save & Continue". But that doesn't quite send as strong of a message to the user as two different buttons that read "Continue to Payment" (before the payment form is visible) and "Save" (after the payment form is visible).

`selectedShippingOption` - this could also be removed, but only if the shipping-option is optional or if we give the shipping-option a preselected default value. Otherwise we'd need to make the shipping-option a part of the form itself, so it would know of an invalid value. It's possible that making the radio a part of the form itself makes the most sense here, but that would mean removing the shipping-options component and manually adding the radios to each form where they are needed.

`updateShippingInfo` - essentially the formSubmitted event.

`continueToPayment` - The purpose of this is purely to avoid calling the ShowPaymentView action every time the shippingInfo is updated, because it only needs to be called the first time. That being said, it wouldn't really be that big of a deal if it was called every time, because it could only be called when the user manually resubmitted the form. It would just clog up the redux actions log a bit. 

I think that `hideContinueToPayment` should definitely stay, unless we're ok with being more vague on the submission button texts. I think that `selectedShippingOption` is kind of ugly, but that the other options are probably worse. And I think `continueToPayment` should probably just be removed, and consumed by `updateShippingInfo` even though it'll mean calling the ShowPaymentView action unnecessarily. What do you think?